src/sasl/cyrsasl_external_verification.erl

```diff
@@ -0,0 +1,47 @@
+%%%=============================================================================
+%%% @copyright (C) 1999-2019, Erlang Solutions Ltd
+%%% @author Denys Gonchar <denys.gonchar@erlang-solutions.com>
+%%% @doc cyrsasl external verification backend
+%%%
+%%% this module is added for demo & testing purposes only.
+%%%
+%%% the next extra fields are added to mongoose_credentials record by cyrsasl_external:
+%%%   * pem_cert        - certificate in PEM format
+%%%   * der_cert        - certificate in DER format
+%%%   * common_name     - CN field (bitstring) of the client's cert (if available)
+%%%   * xmpp_addresses  - list of provided "xmppAddr" fields (bare jids) provided in
+%%%                       the client's certificate (empty list if not available)
+%%%   * auth_id         - authorization identity (bare jid, if provided by the client)
+%%%
+%%% @end
+%%% Created : 17. Apr 2019 11:49
+%%%-------------------------------------------------------------------
+-module(cyrsasl_external_verification).
+-copyright("2018, Erlang Solutions Ltd.").
```

Comment:
2019 ;)

Reply:
that is really old pull request and copy past is evil